hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/CallIdSequenceWithBackpressure.java

```diff
@@ -94,13 +94,20 @@ private boolean hasSpace() {
         return longs.get(INDEX_HEAD) - longs.get(INDEX_TAIL) < maxConcurrentInvocations;
     }
 
-    private void waitForSpace() throws TimeoutException {
-        long deadline = System.nanoTime() + backoffTimeoutNanos;
```

Comment:
Can you restore the deadline? It makes reasoning with timeouts a lot easier.

Reply:
javadoc mentions possibility of negative values for nanoTime call. how will it help knowing deadline value but not knowing what is the start time or the end time, which are not in the exception string. I was thinking that i put the elapsed time in the exception rather than deadline.

Reply:
What you put in the exception is a different issue. This is about the calculation of the deadline: the last possible moment. So if you can restore that logic.

Reply:
@ihsandemir pointed out a bug in this code regarding long overflow and is actually documented in the System.nanoTime. So we need to change this code.